packages/core/contracts-ovm/insured-bridge/implementation/BridgeAdmin.sol

```diff
@@ -126,87 +123,110 @@ contract BridgeAdmin is BridgeAdminInterface, Ownable, OVM_CrossDomainEnabled, L
             _l2Gas,
             abi.encodeWithSignature("setEnableDeposits(address,bool)", _l2Token, _depositsEnabled)
         );
-        emit DepositsEnabled(_l2Token, _depositsEnabled);
+        emit DepositsEnabled(_chainId, _l2Token, _depositsEnabled);
     }
 
     /**
      * @notice Privileged account can associate a whitelisted token with its linked token address on L2. The linked L2
      * token can thereafter be deposited into the Deposit contract on L2 and relayed via the BridgePool contract.
      * @dev Only callable by Owner of this contract. Also initiates a cross-chain call to the L2 Deposit contract to
      * whitelist the token mapping.
+     * @param _chainId L2 network ID where Deposit contract is deployed.
      * @param _l1Token Address of L1 token that can be used to relay L2 token deposits.
      * @param _l2Token Address of L2 token whose deposits are fulfilled by `_l1Token`.
      * @param _bridgePool Address of BridgePool which manages liquidity to fulfill L2-->L1 relays.
      * @param _l2Gas Gas limit to set for relayed message on L2
      */
     function whitelistToken(
+        uint256 _chainId,
         address _l1Token,
         address _l2Token,
         address _bridgePool,
         uint32 _l2Gas
-    ) public onlyOwner depositContractSet nonReentrant() {
+    ) public onlyOwner canRelay(_chainId) nonReentrant() {
         require(_bridgePool != address(0), "BridgePool cannot be zero address");
         require(_l2Token != address(0), "L2 token cannot be zero address");
         require(_getCollateralWhitelist().isOnWhitelist(address(_l1Token)), "Payment token not whitelisted");
 
         require(address(BridgePoolInterface(_bridgePool).l1Token()) == _l1Token, "Bridge pool has different L1 token");
 
-        whitelistedTokens[_l1Token] = L1TokenRelationships({ l2Token: _l2Token, bridgePool: _bridgePool });
+        _whitelistedTokens[_l1Token] = L1TokenRelationships({ l2Token: _l2Token, bridgePool: _bridgePool });
 
-        sendCrossDomainMessage(
-            depositContract,
+        MessengerInterface(_depositContracts[_chainId].messengerContract).relayMessage(
+            _depositContracts[_chainId].depositContract,
             _l2Gas,
             abi.encodeWithSignature("whitelistToken(address,address,address)", _l1Token, _l2Token, _bridgePool)
         );
+        emit WhitelistToken(_chainId, _l1Token, _l2Token, _bridgePool);
+    }
 
-        emit WhitelistToken(_l1Token, _l2Token, _bridgePool);
+    /**************************************
+     *        VIEW FUNCTIONS              *
+     **************************************/
+    function depositContracts(uint256 _chainId) external view override returns (DepositUtilityContracts memory) {
+        return _depositContracts[_chainId];
+    }
+
+    function whitelistedTokens(address _l1Token) external view override returns (L1TokenRelationships memory) {
+        return _whitelistedTokens[_l1Token];
```

Comment:
why not just make the underlying data type public? is there a gas tradeoff with making a function like this rather? I guess it does let you `override` the interface but having a public variable would do the same thing, no?

Reply:
Yeah I couldn't get this to work when using `override` on the `public` mappings. Here's the compile error:

```
TypeError: Overriding public state variable return types differ.
  --> contracts-ovm/insured-bridge/implementation/BridgeAdmin.sol:33:5:
   |
33 |     mapping(address => L1TokenRelationships) public override whitelistedTokens;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Note: Overridden public state variable is here:
  --> contracts-ovm/insured-bridge/interfaces/BridgeAdminInterface.sol:35:5:
   |
35 |     function whitelistedTokens(address) external view returns (L1TokenRelationships memory);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

@mrice32 I recall you might have a solution for this? I think part of the issue is that the interface expects a `struct memory` to be returned, but mappings don't return `memory` types..?